### PR TITLE
Feature/travis ci

### PR DESCRIPTION
**Pull request by gelever**
I am now content w/ this.  Builds, compiles, runs test (including style, valgrind).  The examples are not tested because they take forever to run, but it would not be difficult to add them again.

~This is as far I got.  The dependencies compile and are cached.  So if for some reason they need to be rebuilt, it should be sufficient to delete the install directory.~

~smoothG also compiles.  However, the tests don't do so hot.  Mpich has something going on with executable paths (?).  So the parallel tests don't run.  I couldn't get it to build w/ openmpi.  And openmpi doesn't play nice w/ valgrind.~

- [x] ~fix tests~
- [x] ~parallel tests are super slow~

~using both mpich and openmpi seems to have done something: https://travis-ci.org/LLNL/smoothG/builds/337806566, at least parallel tests run doesn't need both for tests to pass, runs fine w/ mpich for now, but valgrind still broken~

### CODE REVIEW

**Comment by barker29** [comment]
This is a lot of work, @gelever , thank you. For now we can skip the parallel tests and/or the valgrind tests if that makes it easier - I will be happy to just have a CI system that builds and does very basic tests for the moment.

**Comment by gelever** [comment]
@barker29 ok.  I'll try to build mpich to see if it helps with valgrind.  Then we can run some limited number of tests, maybe some sort of `make check`

File: smoothg.astylerc [hunk 1]
```diff
@@ -33,7 +33,7 @@
 --formatted
 #--break-blocks
 --pad-oper
---pad-comma
+#--pad-comma
```
Comment (gelever) [inline smoothg.astylerc]: This doesn't exist in Astyle version 2 which is what is in the ubuntu repo for travis-ci

**Review by barker29 — APPROVED** [review]
Looks good! Thanks for doing this.

**Review by chakshinglee — APPROVED** [review]
Looks great, thanks!

**Pull request merged**